tests/bsim/bluetooth/host/security/security_changed_callback/src/bs_bt_utils.c

```diff
@@ -0,0 +1,197 @@
+#define TEST_TIMEOUT_SIMULATED BS_SECONDS(60)
+
+void test_tick(bs_time_t HW_device_time)
+{
+	bs_trace_debug_time(0, "Simulation ends now.\n");
+	if (bst_result != Passed) {
+		bst_result = Failed;
+		bs_trace_error("Test did not pass before simulation ended.\n");
+	}
+}
+
+void test_init(void)
+{
+	bst_ticker_set_next_tick_absolute(TEST_TIMEOUT_SIMULATED);
+	bst_result = In_progress;
+}
+
+DEFINE_FLAG(flag_is_connected);
+struct bt_conn *g_conn;
+
+void wait_connected(void)
+{
+	LOG_DBG("Wait for connection...");
+	WAIT_FOR_FLAG(flag_is_connected);
+}
+
+void wait_disconnected(void)
+{
+	WAIT_FOR_FLAG_UNSET(flag_is_connected);
+}
+
+static void security_changed(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	LOG_DBG("security changed");
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	UNSET_FLAG(flag_is_connected);
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	ASSERT((!g_conn || (conn == g_conn)), "Unexpected new connection.");
+
+	if (!g_conn) {
+		g_conn = bt_conn_ref(conn);
+	}
+
+	if (err != 0) {
+		clear_g_conn();
+		return;
+	}
+
+	SET_FLAG(flag_is_connected);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+	.security_changed = security_changed,
+};
+
+void clear_g_conn(void)
+{
+	struct bt_conn *conn;
+
+	conn = g_conn;
+	g_conn = NULL;
+	ASSERT(conn, "Test error: No g_conn!\n");
+	bt_conn_unref(conn);
+}
+
+/* The following flags are raised by events and lowered by test code. */
+DEFINE_FLAG(flag_pairing_complete);
+DEFINE_FLAG(flag_bonded);
+DEFINE_FLAG(flag_not_bonded);
+
+static void pairing_complete(struct bt_conn *conn, bool bonded)
+{
+	LOG_ERR("pairing complete");
+	SET_FLAG(flag_pairing_complete);
+
+	if (bonded) {
+		LOG_DBG("Bonded status: true");
+	} else {
+		LOG_DBG("Bonded status: false");
+	}
+
+	if (bonded) {
+		SET_FLAG(flag_bonded);
+	} else {
+		SET_FLAG(flag_not_bonded);
+	}
+}
+
+DEFINE_FLAG(flag_pairing_failed);
+
+static void pairing_failed(struct bt_conn *conn, enum bt_security_err err)
+{
+	LOG_ERR("Pairing failed");
+}
+
+static struct bt_conn_auth_info_cb bt_conn_auth_info_cb = {
+	.pairing_complete = pairing_complete,
+	.pairing_failed = pairing_failed,
+};
+
+void bs_bt_utils_setup(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	ASSERT(!err, "bt_enable failed.\n");
+	err = bt_conn_auth_info_cb_register(&bt_conn_auth_info_cb);
+	ASSERT(!err, "bt_conn_auth_info_cb_register failed.\n");
+
+	err = settings_load();
+	if (err) {
+		FAIL("Settings load failed (err %d)\n", err);
+	}
+}
+
+static void scan_cb_connect_to_first_result__device_found(const bt_addr_le_t *addr, int8_t rssi,
+							  uint8_t type, struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	int err;
+
+	if (g_conn != NULL) {
+		return;
+	}
+
+	/* We're only interested in connectable events */
+	if (type != BT_HCI_ADV_IND && type != BT_HCI_ADV_DIRECT_IND) {
```

Comment:
this doesn't really check that the evt is connectable. 
IMO we know what the peer is going to send, no need to validate, you can just remove this whole `if-else`.